### ISSUE TITLE
CCM >v0.1.0 fails on Photon2 OS

### ISSUE THREAD

**Issue by akutz** [kubernetes/cloud-provider-vsphere#142]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
Running CCM:latest on Photon2 OS results in a panic.

**What you expected to happen**:
No panic.

**How to reproduce it (as minimally and precisely as possible)**:
Use CCM:latest on Photon2 OS.

**Anything else we need to know?**:
When discovering node information, the CCM on Photon2 OS panics:

```shell
2019-02-14T00:28:28.77211952Z stderr F I0214 00:28:28.772000       1 flags.go:27] FLAG: --address="127.0.0.1"
2019-02-14T00:28:28.772228035Z stderr F I0214 00:28:28.772189       1 flags.go:27] FLAG: --allocate-node-cidrs="false"
2019-02-14T00:28:28.772298274Z stderr F I0214 00:28:28.772266       1 flags.go:27] FLAG: --allow-untagged-cloud="false"
2019-02-14T00:28:28.77236737Z stderr F I0214 00:28:28.772325       1 flags.go:27] FLAG: --alsologtostderr="false"
2019-02-14T00:28:28.772413658Z stderr F I0214 00:28:28.772389       1 flags.go:27] FLAG: --bind-address="0.0.0.0"
2019-02-14T00:28:28.772476189Z stderr F I0214 00:28:28.772444       1 flags.go:27] FLAG: --cert-dir="/var/run/kubernetes"
2019-02-14T00:28:28.772532703Z stderr F I0214 00:28:28.772497       1 flags.go:27] FLAG: --cidr-allocator-type="RangeAllocator"
2019-02-14T00:28:28.772613304Z stderr F I0214 00:28:28.772579       1 flags.go:27] FLAG: --cloud-config="/etc/cloud/cloud-provider.conf"
2019-02-14T00:28:28.772668646Z stderr F I0214 00:28:28.772645       1 flags.go:27] FLAG: --cloud-provider="vsphere"
2019-02-14T00:28:28.772731503Z stderr F I0214 00:28:28.772700       1 flags.go:27] FLAG: --cluster-cidr=""
2019-02-14T00:28:28.772775354Z stderr F I0214 00:28:28.772752       1 flags.go:27] FLAG: --cluster-name="kubernetes"
2019-02-14T00:28:28.772841335Z stderr F I0214 00:28:28.772808       1 flags.go:27] FLAG: --concurrent-service-syncs="1"
2019-02-14T00:28:28.772885695Z stderr F I0214 00:28:28.772862       1 flags.go:27] FLAG: --configure-cloud-routes="true"
2019-02-14T00:28:28.772948713Z stderr F I0214 00:28:28.772918       1 flags.go:27] FLAG: --contention-profiling="false"
2019-02-14T00:28:28.772994519Z stderr F I0214 00:28:28.772969       1 flags.go:27] FLAG: --controller-start-interval="0s"
2019-02-14T00:28:28.77306038Z stderr F I0214 00:28:28.773026       1 flags.go:27] FLAG: --feature-gates=""
2019-02-14T00:28:28.773115022Z stderr F I0214 00:28:28.773081       1 flags.go:27] FLAG: --help="false"
2019-02-14T00:28:28.773178693Z stderr F I0214 00:28:28.773146       1 flags.go:27] FLAG: --http2-max-streams-per-connection="0"
2019-02-14T00:28:28.773222896Z stderr F I0214 00:28:28.773199       1 flags.go:27] FLAG: --kube-api-burst="30"
2019-02-14T00:28:28.773286761Z stderr F I0214 00:28:28.773255       1 flags.go:27] FLAG: --kube-api-content-type="application/vnd.kubernetes.protobuf"
2019-02-14T00:28:28.773334447Z stderr F I0214 00:28:28.773308       1 flags.go:27] FLAG: --kube-api-qps="20"
2019-02-14T00:28:28.773395451Z stderr F I0214 00:28:28.773364       1 flags.go:27] FLAG: --kubeconfig="/etc/cloud/kubeconfig"
2019-02-14T00:28:28.773439326Z stderr F I0214 00:28:28.773416       1 flags.go:27] FLAG: --leader-elect="false"
2019-02-14T00:28:28.773503286Z stderr F I0214 00:28:28.773472       1 flags.go:27] FLAG: --leader-elect-lease-duration="15s"
2019-02-14T00:28:28.773547767Z stderr F I0214 00:28:28.773524       1 flags.go:27] FLAG: --leader-elect-renew-deadline="10s"
2019-02-14T00:28:28.773611681Z stderr F I0214 00:28:28.773581       1 flags.go:27] FLAG: --leader-elect-resource-lock="endpoints"
2019-02-14T00:28:28.773655454Z stderr F I0214 00:28:28.773632       1 flags.go:27] FLAG: --leader-elect-retry-period="2s"
2019-02-14T00:28:28.773720892Z stderr F I0214 00:28:28.773688       1 flags.go:27] FLAG: --log-backtrace-at=":0"
2019-02-14T00:28:28.773829952Z stderr F I0214 00:28:28.773775       1 flags.go:27] FLAG: --log-dir=""
2019-02-14T00:28:28.773876082Z stderr F I0214 00:28:28.773852       1 flags.go:27] FLAG: --log-flush-frequency="5s"
2019-02-14T00:28:28.773938182Z stderr F I0214 00:28:28.773908       1 flags.go:27] FLAG: --logtostderr="true"
2019-02-14T00:28:28.773982401Z stderr F I0214 00:28:28.773959       1 flags.go:27] FLAG: --master=""
2019-02-14T00:28:28.774063766Z stderr F I0214 00:28:28.774017       1 flags.go:27] FLAG: --min-resync-period="12h0m0s"
2019-02-14T00:28:28.774113027Z stderr F I0214 00:28:28.774084       1 flags.go:27] FLAG: --node-monitor-period="5s"
2019-02-14T00:28:28.774165879Z stderr F I0214 00:28:28.774136       1 flags.go:27] FLAG: --node-status-update-frequency="5m0s"
2019-02-14T00:28:28.774218087Z stderr F I0214 00:28:28.774194       1 flags.go:27] FLAG: --node-sync-period="0s"
2019-02-14T00:28:28.774277964Z stderr F I0214 00:28:28.774247       1 flags.go:27] FLAG: --port="10253"
2019-02-14T00:28:28.774320546Z stderr F I0214 00:28:28.774298       1 flags.go:27] FLAG: --profiling="false"
2019-02-14T00:28:28.774382383Z stderr F I0214 00:28:28.774352       1 flags.go:27] FLAG: --route-reconciliation-period="10s"
2019-02-14T00:28:28.77442518Z stderr F I0214 00:28:28.774402       1 flags.go:27] FLAG: --secure-port="0"
2019-02-14T00:28:28.774501596Z stderr F I0214 00:28:28.774470       1 flags.go:27] FLAG: --stderrthreshold="2"
2019-02-14T00:28:28.774545089Z stderr F I0214 00:28:28.774522       1 flags.go:27] FLAG: --tls-cert-file=""
2019-02-14T00:28:28.774613902Z stderr F I0214 00:28:28.774577       1 flags.go:27] FLAG: --tls-cipher-suites="[]"
2019-02-14T00:28:28.77465987Z stderr F I0214 00:28:28.774636       1 flags.go:27] FLAG: --tls-min-version=""
2019-02-14T00:28:28.774750594Z stderr F I0214 00:28:28.774691       1 flags.go:27] FLAG: --tls-private-key-file=""
2019-02-14T00:28:28.774792968Z stderr F I0214 00:28:28.774769       1 flags.go:27] FLAG: --tls-sni-cert-key="[]"
2019-02-14T00:28:28.774852869Z stderr F I0214 00:28:28.774826       1 flags.go:27] FLAG: --use-service-account-credentials="true"
2019-02-14T00:28:28.774911876Z stderr F I0214 00:28:28.774869       1 flags.go:27] FLAG: --v="2"
2019-02-14T00:28:28.774957351Z stderr F I0214 00:28:28.774929       1 flags.go:27] FLAG: --version="false"
2019-02-14T00:28:28.775022704Z stderr F I0214 00:28:28.774988       1 flags.go:27] FLAG: --vmodule=""
2019-02-14T00:28:28.778289143Z stderr F W0214 00:28:28.778237       1 authentication.go:55] Authentication is disabled
2019-02-14T00:28:28.778384283Z stderr F I0214 00:28:28.778316       1 insecure_serving.go:49] Serving insecurely on 127.0.0.1:10253
2019-02-14T00:28:28.83044425Z stderr F I0214 00:28:28.830086       1 node_controller.go:89] Sending events to api server.
2019-02-14T00:28:28.840868055Z stderr F I0214 00:28:28.840537       1 pvlcontroller.go:107] Starting PersistentVolumeLabelController
2019-02-14T00:28:28.840914136Z stderr F I0214 00:28:28.840549       1 controller_utils.go:1025] Waiting for caches to sync for persistent volume label controller
2019-02-14T00:28:28.85045527Z stderr F E0214 00:28:28.850239       1 controllermanager.go:240] Failed to start service controller: the cloud provider does not support external load balancers
2019-02-14T00:28:28.850484857Z stderr F I0214 00:28:28.850259       1 controllermanager.go:264] Will not configure cloud provider routes for allocate-node-cidrs: false, configure-cloud-routes: true.
2019-02-14T00:28:28.854729651Z stderr F I0214 00:28:28.854662       1 node_controller.go:329] This node c01.988ba0c.sk8 is registered without the cloud taint. Will not process.
2019-02-14T00:28:28.940747455Z stderr F I0214 00:28:28.940697       1 controller_utils.go:1032] Caches are synced for persistent volume label controller
2019-02-14T00:28:31.715090811Z stderr F I0214 00:28:31.714993       1 node_controller.go:329] This node c01.988ba0c.sk8 is registered without the cloud taint. Will not process.
2019-02-14T00:28:32.915922485Z stderr F E0214 00:28:32.914877       1 connection.go:63] Failed to create govmomi client. err: ServerFaultCode: Cannot complete login due to an incorrect user name or password.
2019-02-14T00:28:33.045949609Z stderr F E0214 00:28:33.045675       1 datacenter.go:107] Unable to find VM by UUID. VM UUID: bd013042-b08d-992f-af31-5c15b5c37e1e
2019-02-14T00:28:33.045969735Z stderr F E0214 00:28:33.045688       1 search.go:170] Error while looking for vm=bd013042-b08d-992f-af31-5c15b5c37e1e(byUUID) in vc=vcenter.sddc-54-70-161-229.vmc.vmware.com and datacenter=SDDC-Datacenter: No VM found
2019-02-14T00:28:33.04597453Z stderr F E0214 00:28:33.045704       1 nodemanager.go:110] WhichVCandDCByNodeId failed. Err: No VM found
2019-02-14T00:28:33.04599851Z stderr F E0214 00:28:33.045845       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
2019-02-14T00:28:33.046010315Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
2019-02-14T00:28:33.046014808Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
2019-02-14T00:28:33.046018915Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
2019-02-14T00:28:33.046022523Z stderr F /home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/asm_amd64.s:522
2019-02-14T00:28:33.046025738Z stderr F /home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/panic.go:513
2019-02-14T00:28:33.046028805Z stderr F /home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/panic.go:82
2019-02-14T00:28:33.046031631Z stderr F /home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/signal_unix.go:390
2019-02-14T00:28:33.046035174Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/nodemanager.go:114
2019-02-14T00:28:33.046038409Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/nodemanager.go:69
2019-02-14T00:28:33.046041743Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/cloud.go:158
2019-02-14T00:28:33.046045236Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/cloud.go:73
2019-02-14T00:28:33.046048002Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/controller.go:195
2019-02-14T00:28:33.046053755Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:554
2019-02-14T00:28:33.04605665Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203
2019-02-14T00:28:33.04605956Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203
2019-02-14T00:28:33.046070896Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:548
2019-02-14T00:28:33.046074413Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
2019-02-14T00:28:33.046077578Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
2019-02-14T00:28:33.046080513Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2019-02-14T00:28:33.046083291Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:546
2019-02-14T00:28:33.046086438Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:390
2019-02-14T00:28:33.046089239Z stderr F /home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71
2019-02-14T00:28:33.046092753Z stderr F /home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/asm_amd64.s:1333
2019-02-14T00:28:33.049215968Z stderr F panic: runtime error: invalid memory address or nil pointer dereference [recovered]
2019-02-14T00:28:33.049239569Z stderr F 	panic: runtime error: invalid memory address or nil pointer dereference
2019-02-14T00:28:33.049243349Z stderr F [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1839d3b]
2019-02-14T00:28:33.049246251Z stderr F 
2019-02-14T00:28:33.049249676Z stderr F goroutine 76 [running]:
2019-02-14T00:28:33.049252851Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
2019-02-14T00:28:33.049259035Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
2019-02-14T00:28:33.049266212Z stderr F panic(0x1b0f300, 0x37ab9d0)
2019-02-14T00:28:33.049269252Z stderr F 	/home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/panic.go:513 +0x1b9
2019-02-14T00:28:33.049282887Z stderr F k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere.(*NodeManager).DiscoverNode(0xc000115ea0, 0xc0004de6c0, 0x24, 0x0, 0x42e286, 0x1e1efa0)
2019-02-14T00:28:33.049286771Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/nodemanager.go:114 +0x11b
2019-02-14T00:28:33.049289834Z stderr F k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere.(*NodeManager).RegisterNode(0xc000115ea0, 0xc000971080)
2019-02-14T00:28:33.049294573Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/nodemanager.go:69 +0x137
2019-02-14T00:28:33.049297605Z stderr F k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere.(*VSphere).nodeAdded(0xc000500ff0, 0x1e99fe0, 0xc000971080)
2019-02-14T00:28:33.049300889Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/cloud.go:158 +0xb2
2019-02-14T00:28:33.049304311Z stderr F k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere.(*VSphere).nodeAdded-fm(0x1e99fe0, 0xc000971080)
2019-02-14T00:28:33.049307197Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/cloud.go:73 +0x3e
2019-02-14T00:28:33.049310057Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(0xc00044bfc0, 0x0, 0xc00044bfd0, 0x1e99fe0, 0xc000971080)
2019-02-14T00:28:33.049313538Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/controller.go:195 +0x49
2019-02-14T00:28:33.04931644Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0x42b3c3, 0xc000858dc8, 0xa0)
2019-02-14T00:28:33.049319879Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:554 +0x21d
2019-02-14T00:28:33.04932285Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0xc000b47e18, 0x42aed2, 0xc00048bd90)
2019-02-14T00:28:33.049325836Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203 +0x9c
2019-02-14T00:28:33.049328468Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
2019-02-14T00:28:33.049331326Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:548 +0x89
2019-02-14T00:28:33.049334028Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000858f68)
2019-02-14T00:28:33.049336756Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
2019-02-14T00:28:33.049339886Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b47f68, 0xdf8475800, 0x0, 0x1ab1001, 0xc000820540)
2019-02-14T00:28:33.04934325Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbe
2019-02-14T00:28:33.049346033Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000858f68, 0xdf8475800, 0xc000820540)
2019-02-14T00:28:33.049348757Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
2019-02-14T00:28:33.049359492Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc0007f1600)
2019-02-14T00:28:33.049362318Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:546 +0x8d
2019-02-14T00:28:33.049365211Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache.(*processorListener).run-fm()
2019-02-14T00:28:33.04937107Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/client-go/tools/cache/shared_informer.go:390 +0x2a
2019-02-14T00:28:33.049375914Z stderr F k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0009a3720, 0xc0006ec9c0)
2019-02-14T00:28:33.049378766Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x4f
2019-02-14T00:28:33.049381527Z stderr F created by k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
2019-02-14T00:28:33.049384732Z stderr F 	/home/travis/gopath/src/k8s.io/cloud-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62
```

I think the issue may be caused by [lines 108-114 in `nodemanager.go`](https://github.com/kubernetes/cloud-provider-vsphere/blob/8f0ef6aca235605cb842a0d4247aa72a27bd8b39/pkg/cloudprovider/vsphere/nodemanager.go#L108-L114) not preventing `vmDI` from being accessed when it is presumably `nil`.

Apparently the following two patches created a merge conflict and the incorrect logic found its way into the final product:

1. Initial Implementation for CSI multi-VC/DC Support - https://github.com/kubernetes/cloud-provider-vsphere/commit/da8021e3116155db5a5323fefa90aa7f4cfa7a49
2. Switch from using glog to klog - https://github.com/kubernetes/cloud-provider-vsphere/commit/0f7fb7458970a02f711b79b6c01b67b47d1dbddb

In `nodemanager.go`, line 110, the error handling in the glog/klog patch doesn’t prevent an error finding the VM from continuing with the function. However, the CSI/DC patch *does*.

The problem may still be a Photon-specific issue, but errors did not used to result in a panic. Now they do. It can be safely assumed that the inability to find a VM causes a secondary effort, which is in fact succeeding on Photon.

I’ve noticed on Photon that `/sys/class/dmi/id/product_uuid` and `/sys/class/dmi/id/product_serial` are both the same endianness:

```shell
[0]root@c01:~$ cat /sys/class/dmi/id/product_uuid
423001BD-8DB0-2F99-AF31-5C15B5C37E1E

[0]root@c01:~$ cat /sys/class/dmi/id/product_serial
VMware-42 30 01 bd 8d b0 2f 99-af 31 5c 15 b5 c3 7e 1e
```

On CentOS and in Debian the files contain UUIDs where one is big endian and the other is little. In fact, there’s code *in* the CCM to translate this. Apparently the result in the CCM on Photon is the UUID in the *correct* format being translated to the *incorrect* format.

Even on CCM:0.1.0 running on Photon2 OS, where the panic does not occur, the initial lookup by UUID fails for the exact same reason:

```shell
2019-02-13T23:27:05.669661653Z stderr F E0213 23:27:05.669474       1 datacenter.go:97] Unable to find VM by UUID. VM UUID: bd013042-b08d-992f-af31-5c15b5c37e1e
2019-02-13T23:27:05.669677637Z stderr F E0213 23:27:05.669485       1 nodemanager.go:241] Error while looking for vm=bd013042-b08d-992f-af31-5c15b5c37e1e(byUUID) in vc=vcenter.sddc-54-70-161-229.vmc.vmware.com and datacenter=SDDC-Datacenter: No VM found
2019-02-13T23:27:05.669688376Z stderr F I0213 23:27:05.669495       1 nodemanager.go:246] Did not find node bd013042-b08d-992f-af31-5c15b5c37e1e in vc=vcenter.sddc-54-70-161-229.vmc.vmware.com and datacenter=SDDC-Datacenter
2019-02-13T23:27:05.715118213Z stderr F I0213 23:27:05.715037       1 nodemanager.go:280] Found node c01.988ba0c.sk8 as vm=VirtualMachine:vm-16044 in vc=vcenter.sddc-54-70-161-229.vmc.vmware.com and datacenter=SDDC-Datacenter
2019-02-13T23:27:05.715204209Z stderr F I0213 23:27:05.715151       1 nodemanager.go:282] Hostname: c01.988ba0c.sk8 UUID: 423001bd-8db0-2f99-af31-5c15b5c37e1e
2019-02-13T23:27:05.71527719Z stderr F I0213 23:27:05.715234       1 instances.go:114] instances.InstanceID() FOUND with c01.988ba0c.sk8
2019-02-13T23:27:05.715358423Z stderr F I0213 23:27:05.715307       1 instances.go:79] instances.NodeAddressesByProviderID() CACHED with 423001bd-8db0-2f99-af31-5c15b5c37e1e
```

The only difference is that in this case no panic occurs and thus the VM is resolved by name.

I think it’s pretty clear what the problem is. I also think the bug introduced recently only reveals an underlying problem with Photon2 OS -- the two files should not be the same endianness. At least that’s not how both CentOS and Debian behave. For example, here is CentOS:

```shell
[0]root@c01-ccm-ci-latest-adc83b1-86:~$ cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

[0]root@c01-ccm-ci-latest-adc83b1-86:~$ uname -a
Linux c01-ccm-ci-latest-adc83b1-86.vmware.ci 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

[0]root@c01-ccm-ci-latest-adc83b1-86:~$ cat /sys/class/dmi/id/product_serial 
VMware-42 30 97 bd 95 b0 be 1e-22 d2 fc 82 c4 46 9a 90

[0]root@c01-ccm-ci-latest-adc83b1-86:~$ cat /sys/class/dmi/id/product_uuid 
BD973042-B095-1EBE-22D2-FC82C4469A90
```

While `42 30 97 bd 95 b0 be 1e-22 d2 fc 82 c4 46 9a 90` and `BD973042-B095-1EBE-22D2-FC82C4469A90` may look different, they're the same value formatted differently. This is what the CCM expects. Because the CCM code expects this, on Photon2 OS the CCM is unknowingly converting a *valid* UUID into an invalid one.

**Environment**:
- vsphere-cloud-controller-manager version: 
```
gcr.io/cloud-provider-vsphere/vsphere-cloud-controller-manager@sha256:deba22958d2d0556b726ee4b4fb75c0d708f0d07b004f705fef7e2262af4974b
```
- OS (e.g. from /etc/os-release):
```shell
NAME="VMware Photon OS"
VERSION="2.0"
ID=photon
VERSION_ID=2.0
PRETTY_NAME="VMware Photon OS/Linux"
ANSI_COLOR="1;34"
HOME_URL="https://vmware.github.io/photon/"
BUG_REPORT_URL="https://github.com/vmware/photon/issues"
```
- Kernel (e.g. `uname -a`):
```shell
Linux c01.988ba0c.sk8 4.9.140-3.ph2-esx #1-photon SMP Sat Jan 19 14:32:14 UTC 2019 x86_64 Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz GenuineIntel GNU/Linux
```

- Install tools:
sk8

- Others:

**Comment by dvonthenen** [comment]
I did a fresh install of photon 2 and 3 and it looks like Photon 2 behaves differently when reporting the UUID. Photon 3 conforms to how RHEL, CentOS, Ubuntu, and etc reports the UUID. Looking into how I can go about supporting this.

**Comment by frapposelli** [comment]
this should be fixed by #143 

@akutz can you verify?

**Comment by dvonthenen** [comment]
Verbal verify in slack

/close
